addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java

```diff
@@ -219,51 +220,57 @@ public HomematicGateway getGateway() {
     }
 
     /**
-     * Updates the thing for the given Homematic device.
+     * Updates all things for the given Homematic device.
      */
-    private void updateThing(HmDevice device) {
-        Thing hmThing = getThingByUID(UidUtils.generateThingUID(device, getThing()));
-        if (hmThing != null && hmThing.getHandler() != null) {
-            HomematicThingHandler thingHandler = (HomematicThingHandler) hmThing.getHandler();
-            thingHandler.thingUpdated(hmThing);
-            for (Channel channel : hmThing.getChannels()) {
-                thingHandler.handleRefresh(channel.getUID());
+    private void updateThings(HmDevice device) {
```

Comment:
That looks like an EXTREMELY costly operation - rather a bad smell, if you ask me...